packages/app/src/specs/generators/ExpandableFileChooser.vue

```diff
@@ -0,0 +1,172 @@
+<!--
+/**==============================================
+ * *          ExpandableFileChooser.vue
+ * Filter a list of files by a mix of glob pattern
+ * and a search string that includes a file's relative
+ * path.
+ *
+ * When clicked, the files will expand to display the
+ * #expanded-content slot.
+ *
+ * Features to note: debouncing + loading + row expanding
+ * ==============================================
+ * *                Debouncing
+ *
+ * ? To prevent calling update too frequently we debounce
+ * ? both the "No Results" UI and the update event
+ *
+ * ============================================**/
+-->
+
+<template>
+  <CreateSpecModalBody
+    variant="bare"
+    class="bg-white flex flex-col px-24px
+  relative"
+  >
+    <FileMatch
+      ref="fileMatchRef"
+      v-model:pattern="filePathSearch"
+      v-model:extensionPattern="localExtensionPattern"
+      class="bg-white pt-24px pb-12px top-0px z-10 sticky"
+      :matches="matches"
+    >
+      <template
+        v-if="loading"
+        #matches
+      >
+        <i-cy-loading_x16 class="h-24px mr-10px animate-spin w-24px" />
+      </template>
+    </FileMatch>
+
+    <div
+      v-show="loading"
+      data-testid="loading"
+    >
+      <!-- TODO(ryan) UNIFY-865: Get mocks for a loading state here -->
+      Loading
+    </div>
+    <ExpandableFileList
+      v-show="!loading"
+      :style="{ paddingTop: `${fileMatchHeight + 36}px` }"
+      class="right-24px left-24px absolute"
+      :files="filteredFiles"
+      :search="filePathSearch"
+    >
+      <template #expanded-content="{file}">
+        <ReactComponentList
```

Comment:
I want to re-work this so that the expandable section can be passed as a prop into `ExpandableFileChooser` but I couldn't figure out how to get the slots to work that way. 

That way this can be generic if we want to use it again for something else

Reply:
We can sync on this, but this seems fine to me? What would you like to change?

Reply:
I wanted to make it more generic. Right now, if you use `ExpandableFileChooser`, the expanded content when you click a file is always `ReactComponentList`. I wanted the caller of `ExpandableFileChooser` to be able to pass in the component that it would like to be rendered when a row in `ExpandableFileList` is expanded

Reply:
What about this: https://github.com/cypress-io/cypress/pull/25079